sklearn/utils/estimator_checks.py

```diff
@@ -403,6 +404,40 @@ def pairwise_estimator_convert_X(X, estimator, kernel=linear_kernel):
     return X
 
 
+def _generate_sparse_matrix(X_csr):
+    """Generate matrices in multiple formats for CSR,CSC and COO matrices
+
+        Parameters
+        ----------
+
+        X_csr: CSR Matrix
+            Input matrix in CSR format
+
+        Returns
+        -------
+
+        out: iter(Matrices)
+            In format['dok', 'lil', 'dia', 'bsr', 'csr', 'csc', 'coo',
+             'coo_64', 'csc_64', 'csr_64']
+    """
+
+    for sparse_format in ['dok', 'lil', 'dia', 'bsr', 'csr', 'csc', 'coo']:
+        yield sparse_format, X_csr.asformat(sparse_format)
```

Comment:
Could we add a `X_csr.copy().asformat` here (and on L430) ?

Just to be sure that we are not taking in e.g. a CSR matrix, [returning it as it is](https://github.com/scipy/scipy/blob/80e76d43f0ada0a1964195cb6d13ae519feb07e8/scipy/sparse/base.py#L315-L319) when `sparse_format="csr"` that get's then in place modified by some estimator and so the next iteration of `sparse_format="csc"` would use this modified version.

Reply:
There should be a copy=True kwarg iirc